pywhy_stats/discrepancy/base.py

```diff
@@ -0,0 +1,127 @@
+from typing import Callable, Optional
```

Comment:
These should be able to be used for all discrepancy tests since testing `P(y|x) = P'(y|x)` usually requires some type of "matching" done on the covariates, which in this case is done via the propensity scores.